lib/engines/hcktest/tests.rb

```diff
@@ -132,11 +132,36 @@ def wait_queued_test(id)
       end
     end
 
-    def test_parameters(test_name)
+    def select_test_config(test_name, config)
       @project.engine.config.tests_config
               .select { _1.tests.include?(test_name) }
-              .flat_map(&:parameters)
-              .to_h { |parameter| [parameter.name, parameter.value] }
+              .flat_map(&config)
+    end
+
+    def test_parameters(test_name)
+      select_test_config(test_name, :parameters)
+        .to_h { |parameter| [parameter.name, parameter.value] }
+    end
+
+    def run_guest_command(test, command, desc)
+      @logger.info("Running command (#{desc}) on client #{@client.name}")
+      @tools.run_on_machine(@client.name, command, command)
+
+      return unless support_needed?(test)
+
+      @logger.info("Running command (#{desc}) on client #{@support.name}")
+      @tools.run_on_machine(@support.name, command, command)
```

Comment:
This method has code duplication for `@client` and `@support`. It is better to have a method that runs a command on one client instead of this to reduce redundancy.